### PR TITLE
Make PackageRepo.Dir() work with a GOPATH with multiple entries.

### PR DESCRIPTION
**Pull request by joho**
Like it says in the title.

### CODE REVIEW

**Comment by joho** [comment]
don't merge yet. slightly doesn't work in all cases.

**Comment by mtibben** [comment]
Looks good. Instead of project call it repo

**Comment by joho** [comment]
Ended up removing that panic... because it panicked when I was installing a repo/package that wasn't present.

File: vcs/gogpm.go [hunk 1]
```diff
@@ -73,7 +73,26 @@ func (p *PackageRepo) RootImportPath() string {
...
+	// construct path options for repo
+	fullPaths := []string{}
+	for _, path := range paths {
+		fullPaths = append(fullPaths, filepath.Join(path, "src", p.rr.root))
+	}
+
+	// return first instance where lib exists
+	for _, path := range fullPaths {
+		_, err := os.Stat(path)
+		if err == nil {
```
Comment (mtibben) [inline vcs/gogpm.go]: do you need to [`IsDir`](http://golang.org/pkg/os/#FileMode.IsDir) here ?

Comment (joho) [inline vcs/gogpm.go]: probably

File: vcs/gogpm.go [hunk 2]
```diff
@@ -73,7 +74,30 @@ func (p *PackageRepo) RootImportPath() string {
 }
 
 func (p *PackageRepo) Dir() string {
-	return filepath.Join(os.Getenv("GOPATH"), "src", p.rr.root)
+	// split gopath
+	goPath := os.Getenv("GOPATH")
+	paths := strings.Split(goPath, ":")
```
Comment (shurcooL) [inline vcs/gogpm.go]: Hi,

This is not ideal.

Instead of `strings.Split` you should use [`filepath.SplitList`](https://godoc.org/path/filepath#SplitList) which is made for this purpose. The reason is that ":" is only going to work on some systems, but not all:

> The GOPATH environment variable lists places to look for Go code. On Unix, the value is a colon-separated string. On Windows, the value is a semicolon-separated string. On Plan 9, the value is a list.

Comment (joho) [inline vcs/gogpm.go]: Thanks @shurcooL. I wasn't aware of that platform difference (and the helper func), will put in another PR to fix.
